### ISSUE TITLE
Subprocess of cellranger freezes which causes everything to wait for that process.

### ISSUE THREAD

**Issue by MarijnBerg** [LungCellAtlas/scRNAseq_pipelines#9]
I'm running the pipeline on a slurm cluster, making a job of the whole analysis and running it on a node with local settings.

The testdata worked fine as did a test dataset I ran earlier but five other samples I tried to process all stopped working at around the same point.

The process that hangs is one of the BUCKET_BY_BC chunk processing parts. This causes everything else to wait for this to complete until the allotted computation time runs out and the whole job is killed. The individual process keeps running though and the cluster manager had to manually kill them.

The data is all Single Cell 3' V2 data.

Has anyone else encountered similar issues? Any ideas on what this issue might stem from? I've not had something like this happen when running cellranger locally.

**Comment by LisaSikkema** [comment]
Hey Marijn, thanks for posting this!
@wathom Any idea what could be the issue?

**Comment by LisaSikkema** [comment]
Hi Marijn, 
The problem doesn't sound familiar to either Thomas or me. It might just be related to cellranger. 
Have you tried to run a second time, and does the problem re-occur?

**Comment by MarijnBerg** [comment]
I initially tried it on multiple separate jobs, several had the same or similar issues. I canceled the other jobs because I expected them to fail too and didn't want to hog the cluster.

Two of the jobs failed by freezing in the detect chemistry phase so I now set the chemistry manually in the tsv file and I'll see if it works on one sample. Cellranger seems to have worked fine there so I'll try running some of the other samples that previously failed and see what happens.

And it does sound like a cellranger issue but I'm short on ideas on how to find out exactly what is/was going wrong here to avoid it in the future.

**Comment by MarijnBerg** [comment]
It seems to be running fine now that I explicitly set the chemistry in the TSV files rather than rely on cellranger to detect it but it's still weird that this pops up.

**Comment by LisaSikkema** [comment]
Okay thanks for checking and updating us!

**Comment by MarijnBerg** [comment]
Ok, that solution doesn't actually seem to have solved the whole problem.

I have a few samples now that failed in the pipeline on the cluster and reproducibly fail in the pipeline on a local installation.

It is an issue with cellranger, that's where the pipeline fails and when I run cellranger from the pipeline conda environment separately it also fails at the same spot. I tried reinstalling cellranger 3.1 in the conda environment to make sure it wasn't a botched installation and it still happens. Some other samples and the test dataset work fine.

Running the same sample through cellranger 4.0 I have installed on the same computer and that works fine.

I checked the fastq files with fastq_utils and they seem fine.

How that relates to the cluster specific issue of freezing nodes is still a mystery but will probably not be an issue if cellranger doesn't fail.

Any suggestions on how to proceed?
